commit 78f1ddbb498283c2445c11b0dfa666424c301803
Author: Theodore Ts'o <tytso@mit.edu>
Date:   Mon Jul 27 23:09:47 2009 -0400

    ext4: Avoid null pointer dereference when decoding EROFS w/o a journal
    
    We need to check to make sure a journal is present before checking the
    journal flags in ext4_decode_error().
    
    Signed-off-by: Eric Sesterhenn <eric.sesterhenn@lsexperts.de>
    Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>

Adjusted to apply to Debian's 2.6.24 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.24.orig/fs/ext4/super.c linux-source-2.6.24/fs/ext4/super.c
--- linux-source-2.6.24.orig/fs/ext4/super.c	2008-01-24 15:58:37.000000000 -0700
+++ linux-source-2.6.24/fs/ext4/super.c	2010-01-31 21:44:20.000000000 -0700
@@ -256,7 +256,8 @@ static const char *ext4_decode_error(str
 		errstr = "Out of memory";
 		break;
 	case -EROFS:
-		if (!sb || EXT4_SB(sb)->s_journal->j_flags & JBD2_ABORT)
+		if (!sb || (EXT4_SB(sb)->s_journal &&
+			    EXT4_SB(sb)->s_journal->j_flags & JBD2_ABORT))
 			errstr = "Journal has aborted";
 		else
 			errstr = "Readonly filesystem";
